src/uct/ib/mlx5/rc/rc_mlx5_devx.c

```diff
@@ -243,6 +243,15 @@ uct_rc_mlx5_devx_init_rx_common(uct_rc_mlx5_iface_common_t *iface,
 
     iface->rx.srq.db = &iface->rx.srq.devx.dbrec->db[MLX5_RCV_DBR];
 
+    if (iface->config.srq_topo == UCT_RC_MLX5_SRQ_TOPO_AUTO) {
+        ddp_enabled            = (md->dp_ordering_cap.rc ==
+                                  UCT_IB_MLX5_DP_ORDERING_OOO_ALL) ||
+                                 (md->dp_ordering_cap.dc ==
+                                  UCT_IB_MLX5_DP_ORDERING_OOO_ALL);
+        iface->config.srq_topo = ddp_enabled ? UCT_RC_MLX5_SRQ_TOPO_LIST :
+                                               UCT_RC_MLX5_SRQ_TOPO_CYCLIC;
```

Comment:
srq_topo is currently a priority-based list. so we should say "cyclic" is not supported with DDP and move to the next priority  - instead of adding new "auto" value